pkg/operator/ceph/config/keyring/admin.go

```diff
@@ -48,7 +52,22 @@ func (s *SecretStore) Admin() *AdminStore {
 }
 
 // CreateOrUpdate creates or updates the admin keyring secret with cluster information.
-func (a *AdminStore) CreateOrUpdate(c *cephclient.ClusterInfo) error {
+func (a *AdminStore) CreateOrUpdate(c *cephclient.ClusterInfo, context *clusterd.Context, annotation v1.AnnotationsSpec) error {
 	keyring := fmt.Sprintf(adminKeyringTemplate, c.CephCred.Secret)
-	return a.secretStore.CreateOrUpdate(adminKeyringResourceName, keyring)
+	err := a.secretStore.CreateOrUpdate(adminKeyringResourceName, keyring)
+	if err != nil {
+		return err
+	}
+	// Update `rook-ceph-admin-keyring` secret annotation
+	secret, err := context.Clientset.CoreV1().Secrets(c.Namespace).Get(c.Context, keyringSecretName(adminKeyringResourceName), metav1.GetOptions{})
```

Comment:
These lines 62-67 are basically the same in two places. How about factoring out to a helper method such as `ApplyClusterMetadataToSecret()` that takes the secret name?

Reply:
done.